### PR TITLE
feat(s3): Adding s3 upload code

### PR DESCRIPTION
**Pull request by fkesheh**
## Summary by CodeRabbit

* **New Features**
  * Optional S3-backed file storage (feature flag) with presigned upload/download support, on-demand URL resolution, and image URL caching/prefetching for faster previews and fewer re-fetches.
  * UI file parts no longer surface stored URLs; downloads/previews resolve URLs on demand and surface S3 identifiers where applicable.

* **Tests**
  * Added comprehensive tests and mocks covering S3 upload/download, presigned URL generation, deletion/cleanup, batch URL fetching, and edge cases.

* **Chores**
  * Added AWS SDK deps, S3 config env examples, and an S3 security validation script.

### CODE REVIEW

File: lib/ai/tools/utils/sandbox-file-uploader.ts [hunk 1]
```diff
@@ -4,8 +4,10 @@ import { ConvexHttpClient } from "convex/browser";
 import { api } from "@/convex/_generated/api";
 import { Id } from "@/convex/_generated/dataModel";
 import type { Sandbox } from "@e2b/code-interpreter";
+import { generateS3UploadUrl } from "@/convex/s3Utils";
 
 const DEFAULT_MEDIA_TYPE = "application/octet-stream";
+const USE_S3_STORAGE = process.env.NEXT_PUBLIC_USE_S3_STORAGE === "true";
```
[the remark this answers is not in the document]
Comment (fkesheh) [reply lib/ai/tools/utils/sandbox-file-uploader.ts:10]: Not needed, front and back should be in sync

**Comment by rostikmanko** [comment]
✅ **File Management Functionality Verified**

I've thoroughly tested the file handling features with the following results:
- Image, TXT and PDF uploads: ✅ Working correctly
- AI can see and analyze files: ✅ Confirmed working
- File deletion via chatbot interface: ✅ Working as expected
- File deletion through chat/message editing: ✅ Functioning properly
- User downloading of own files and assistant files via message UI: ✅ Working correctly
- Legacy file accessibility: ✅ Old files remain accessible to both AI and users through the message UI download feature
- Temporary chat file functionality: ✅ AI can see files and users can install files in temp chats

**Code Review Finding:**
While examining the codebase, I identified that the `share_id` and `share_date` fields were added to the chat schema:
```javascript
share_id: v.optional(v.string()),
share_date: v.optional(v.number()),
```
These fields appear to be related to chat sharing functionality that hasn't been pushed to production yet. I've removed them to prevent unnecessary code bloat and potential confusion.
